src/main/scala/mesosphere/marathon/MesosHeartbeatMonitor.scala

```diff
@@ -0,0 +1,160 @@
+package mesosphere.marathon
+
+import java.util.UUID
+import javax.inject.{ Inject, Named }
+
+import akka.actor.ActorRef
+import mesosphere.marathon.util.heartbeat.Heartbeat
+import org.apache.mesos.{ Scheduler, SchedulerDriver }
+import org.apache.mesos.Protos._
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConverters._
+
+// @constructor create a mesos Scheduler decorator that intercepts callbacks from a mesos SchedulerDriver,
+// translating each callback into some kind of HeartbeatMonitor.Message that is asynchronously delivered to a
+// heartbeatActor. All callback parameters are passed through, unchanged, to the given delegate scheduler
+// implementation.
+//
+// @param scheduler is the delegate scheduler implementation
+// @param heartbeatActor is the receipient of generated Heartbeat.Message's
+//
+// @see mesosphere.util.monitor.HeartbeatMonitor
+// @see org.apache.mesos.Scheduler
+// @see org.apache.mesos.SchedulerDriver
+class MesosHeartbeatMonitor @Inject() (
+    @Named(MesosHeartbeatMonitor.BASE) scheduler: Scheduler,
+    @Named(ModuleNames.MESOS_HEARTBEAT_ACTOR) heartbeatActor: ActorRef
+) extends Scheduler {
+
+  import MesosHeartbeatMonitor._
+
+  private[this] val log = LoggerFactory.getLogger(getClass.getName)
+
+  log.debug(s"created mesos heartbeat monitor for scheduler $scheduler")
+
+  protected[marathon] def heartbeatReactor(driver: SchedulerDriver): Heartbeat.Reactor = new Heartbeat.Reactor {
+    // virtualHeartbeatTasks is sent in a reconciliation message to mesos in order to force a
+    // predictable response: the master (if we're connected) will send back a TASK_LOST because
+    // the fake task ID and agent ID that we use will never actually exist in the cluster.
+    // this is part of a short-term workaround: will no longer be needed once marathon is ported
+    // to use the new mesos v1 http API.
+    private[this] val virtualHeartbeatTasks: java.util.Collection[TaskStatus] = Seq(fakeHeartbeatStatus).asJava
+
+    override def onSkip(): Unit = {
+      log.debug("Prompting mesos for a heartbeat via explicit task reconciliation")
+      driver.reconcileTasks(virtualHeartbeatTasks)
```

Comment:
@aquamatthias If a framework tries to kill an unknown task, the master will automatically trigger an implicit reconciliation for all of the framework's tasks. That seems like weird behavior to me, not sure why we do that.

If you want to trigger a response from Mesos, doing explicit reconciliation for an unknown agent/task ID seems fine to me -- you will get TASK_LOST (or TASK_UNKNOWN in the future), but won't trigger any automatic reconciliation.